composeApp/src/commonMain/kotlin/sp/bvantur/tasky/event/presentation/CreateEventViewModel.kt

```diff
@@ -0,0 +1,70 @@
+package sp.bvantur.tasky.event.presentation
+
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+import sp.bvantur.tasky.core.domain.DispatcherProvider
+import sp.bvantur.tasky.core.presentation.SingleEventHandler
+import sp.bvantur.tasky.core.presentation.SingleEventHandlerImpl
+import sp.bvantur.tasky.core.presentation.TextData
+import sp.bvantur.tasky.core.presentation.ViewModelUserActionHandler
+import sp.bvantur.tasky.core.presentation.ViewStateViewModel
+import sp.bvantur.tasky.event.ui.model.CreateEventModel
+import sp.bvantur.tasky.event.ui.model.SingleInputModel
+
+class CreateEventViewModel(dispatcherProvider: DispatcherProvider) :
+    ViewStateViewModel<CreateEventViewState>(
+        initialViewState = CreateEventViewState(),
+        dispatcherProvider = dispatcherProvider
+    ),
+    ViewModelUserActionHandler<CreateEventUserAction>,
+    SingleEventHandler<CreateEventSingleEvent> by SingleEventHandlerImpl(dispatcherProvider) {
+
+    fun onLoadInitialData(eventData: CreateEventModel) {
+        val title = eventData.title?.let {
+            TextData.DynamicString(it)
+        } ?: viewStateFlow.value.title
+
+        val description = eventData.description?.let {
+            TextData.DynamicString(it)
+        } ?: viewStateFlow.value.description
+
+        viewModelScope.launch {
+            emitViewState { viewState ->
+                viewState.copy(title = title, description = description)
+            }
+        }
+    }
```

Comment:
Instead of passing this data (title, description) from `SavedStateHandle` into a model and then to your viewmodel, could it not make more sense to access the data directly from the `SavedStateHandle` from inside your viewmodel?

```
class CreateEventViewModel (
    // …
    savedStateHandle: SavedStateHandle
) : ViewModel {
    val title = savedStateHandle[“key"]
}
```

You could also get a `StateFlow`from the saved state handle:

```
savedStateHandle.getStateFlow<String?>(key = “key”, initialValue = null)
```